x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/users_table.tsx

```diff
@@ -0,0 +1,140 @@
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { upperFirst } from 'lodash';
+
+import { EuiBadge, EuiBasicTableColumn, EuiInMemoryTable, EuiTextColor } from '@elastic/eui';
+
+import { ASRoleMapping } from '../../app_search/types';
+import { SingleUserRoleMapping } from '../../shared/types';
+import { WSRoleMapping } from '../../workplace_search/types';
+
+import { INVITATION_PENDING_LABEL } from './constants';
+
+import { UsersAndRolesRowActions } from './';
+
+interface AccessItem {
+  name: string;
+}
+
+interface SharedUser extends SingleUserRoleMapping<ASRoleMapping | WSRoleMapping> {
+  accessItems: AccessItem[];
+  username: string;
+  email: string | null;
+  roleType: string;
+  id: string;
+}
+
+interface SharedRoleMapping extends ASRoleMapping, WSRoleMapping {
+  accessItems: AccessItem[];
+}
+
+interface Props {
+  accessItemKey: 'groups' | 'engines';
+  singleUserRoleMappings: Array<SingleUserRoleMapping<ASRoleMapping | WSRoleMapping>>;
+  initializeSingleUserRoleMapping(roleId: string): string;
+  handleDeleteMapping(roleId: string): string;
+}
+
+const noItemsPlaceholder = <EuiTextColor color="subdued">&mdash;</EuiTextColor>;
+const invitationBadge = <EuiBadge color="hollow">{INVITATION_PENDING_LABEL}</EuiBadge>;
+
+export const UsersTable: React.FC<Props> = ({
+  accessItemKey,
+  singleUserRoleMappings,
+  initializeSingleUserRoleMapping,
+  handleDeleteMapping,
+}) => {
+  // 'accessItems' is needed because App Search has `engines` and Workplace Search has `groups`.
+  const users = ((singleUserRoleMappings as SharedUser[]).map((user) => ({
+    username: user.elasticsearchUser.username,
+    email: user.elasticsearchUser.email,
+    roleType: user.roleMapping.roleType,
+    id: user.roleMapping.id,
+    accessItems: (user.roleMapping as SharedRoleMapping)[accessItemKey],
+    invitation: user.invitation,
+  })) as unknown) as Array<Omit<SharedUser, 'elasticsearchUser | roleMapping'>>;
+
+  const columns: Array<EuiBasicTableColumn<SharedUser>> = [
+    {
+      field: 'username',
+      name: 'Username',
+      render: (_, { username }: SharedUser) => username,
+    },
+    {
+      field: 'email',
+      name: 'Email',
+      render: (_, { email, invitation }: SharedUser) => {
+        if (!email) return noItemsPlaceholder;
+        return (
+          <div data-test-subj="EmailDisplayValue">
+            {email} {invitation && invitationBadge}
+          </div>
+        );
+      },
+    },
+    {
+      field: 'roleType',
+      name: 'Role',
+      render: (_, user: SharedUser) => user.roleType,
+    },
+    {
+      field: 'accessItems',
+      name: upperFirst(accessItemKey),
+      render: (_, { accessItems }: SharedUser) => {
+        // Design calls for showing the first 2 items followed by a +x after those 2.
+        // ['foo', 'bar', 'baz'] would display as: "foo, bar + 1"
+        const numItems = accessItems.length;
+        if (numItems === 0) return <span data-test-subj="AllItems">All</span>;
```

Comment:
i18n fixed in a later commit (https://github.com/elastic/kibana/pull/102826/commits/beed0f733a7671f208bf8879b11767c3c6d2e691)